Brokerages/Samco/SamcoBrokerage.cs

```diff
@@ -0,0 +1,1072 @@
+        /// </summary>
+        /// <param name="symbols">The list of symbols to subscribe</param>
+        public void Subscribe(IEnumerable<Symbol> symbols)
+        {
+            if (symbols.Count() <= 0)
+            {
+                return;
+            }
+            var sub = new Subscription();
+            foreach (var symbol in symbols)
+            {
+
+                try
+                {
+                    var quote = GetQuote(symbol);
+                    _subscriptionsById[quote.listingId] = symbol;
+                    sub.request.data.symbols.Add(new Subscription.Symbol { symbol = quote.listingId });
+                    if (!subscribeInstrumentTokens.Contains(quote.listingId))
+                    {
+                        subscribeInstrumentTokens.Add(quote.listingId);
+
+                        unSubscribeInstrumentTokens.Remove(quote.listingId);
+                        _subscriptionsById[quote.listingId] = symbol;
+                    }
+                }
+                catch (Exception exception)
+                {
+                    Log.Error(exception);
+                    throw;
+                }
+            }
+            var request = JsonConvert.SerializeObject(sub);
+            WebSocket.Send(request);
+            WebSocket.Send("\n");
+
+        }
+
+
+        /// <summary>
+        /// Connects to Samco Websocket
+        /// </summary>
+        public override void Connect()
+        {
+            if (IsConnected)
+                return;
+
+            Log.Trace("SamcoBrokerage.Connect(): Connecting...");
+
+            var resetEvent = new ManualResetEvent(false);
+            EventHandler triggerEvent = (o, args) => resetEvent.Set();
+            WebSocket.Open += triggerEvent;
+            WebSocket.Connect();
+            if (!resetEvent.WaitOne(ConnectionTimeout))
+            {
+                throw new TimeoutException("Websockets connection timeout.");
+            }
+            WebSocket.Open -= triggerEvent;
+        }
+
+        /// <summary>
+        /// Closes the websockets connection
+        /// </summary>
+        public override void Disconnect()
+        {
+            if (WebSocket.IsOpen)
+            {
+                WebSocket.Close();
+            }
+        }
+
+
+        /// <summary>
+        /// Places a new order and assigns a new broker ID to the order
+        /// </summary>
+        /// <param name="order">The order to be placed</param>
+        /// <returns>True if the request for a new order has been placed, false otherwise</returns>
+        public override bool PlaceOrder(Order order)
+        {
+            var submitted = false;
+
+            _messageHandler.WithLockedStream(() =>
+            {
+                var security = _securityProvider.GetSecurity(order.Symbol);
+                var orderFee = security.FeeModel.GetOrderFee(new OrderFeeParameters(security, order));
```

Comment:
The order fee should come with the filling not the submitting of the orders, right? same for order update